addons/test_mail/tests/test_mail_gateway.py

```diff
@@ -442,6 +442,21 @@ def test_message_process_email_author_exclude_alias(self):
         self.assertFalse(record.message_ids[0].author_id)
         self.assertEqual(record.message_ids[0].email_from, from_1.email_formatted)
 
+    def test_message_parse_invalid_encoding_in_attachment(self):
+        """ Incoming email containing an xml attachment with unknown characters (�) should not raise an Exception when
+        encoding using the charset of the email.
+        """
+        with self.assertLogs('odoo.addons.mail.models.mail_thread', level="WARNING") as capture:
+            record = self.format_and_process(test_mail_data.MAIL_MULTIPART_INVALID_ENCODING, self.email_from, 'groups@test.com')
+
+        self.assertEqual(capture.output, [
+            ("WARNING:odoo.addons.mail.models.mail_thread:Failed to encode the content of the attachment with error: "
+             "'ascii' codec can't encode characters in position 177-182: ordinal not in range(128)"),
+        ])
+        self.assertEqual(record.message_main_attachment_id.name, 'bis3_with_error_encoding_address.xml')
+        # the invalid character "é" was ignored
+        self.assertIn("Chausse de Bruxelles", record.message_main_attachment_id.raw.decode())
+
```

Comment:
I don't agree with this test, unknown characters should nor be discarded silently (`'ignore'`) nor be replaced with tofu (`'replace'`). Imagine an email that is mostly written using a Cyrillic/Arab/CJK script but having a wrong encoding declared on the envelop, the email would land in Odoo containing mostly blanks or tofu, I don't think this is ok.

My opinion on this is that we should not process this email, it is obviously wrong and needs to be fixed at the source: inside of the software that generated it. My opinion too is that there should be a clear error reporting to the user, maybe an automatic email reply containing an error report.

From my understanding of our email processing, the problem here is that errors only appear in the logs and are not reported to the users in a friendly way.